### PR TITLE
Migrate pull-secrets-store-csi-driver-e2e-vault (csi job) to community cluster

### PR DESCRIPTION
**Pull request by SD-13**
Migrate pull-secrets-store-csi-driver-e2e-vault to community clusters
#31792 
/cc @rjsadow @ameukam

### CODE REVIEW

**Comment by SD-13** [comment]
@rjsadow @ameukam Can we know how much resource required for a particular job? Is there a way to know the usage from past runs, both in GKE and EKS?

File: config/jobs/kubernetes-sigs/secrets-store-csi-driver/secrets-store-csi-driver-config.yaml [hunk 1]
```diff
@@ -163,6 +164,13 @@ presubmits:
             make e2e-bootstrap e2e-helm-deploy e2e-vault
         securityContext:
           privileged: true
+        resources:
+          requests:
+            cpu: "4"
+            memory: "8Gi"
```
Comment (SD-13) [inline config/jobs/kubernetes-sigs/secrets-store-csi-driver/secrets-store-csi-driver-config.yaml:170]: I think I allocated more memory than required, what value should I use, maybe 4 or 6 Gi?

Comment (ameukam) [reply config/jobs/kubernetes-sigs/secrets-store-csi-driver/secrets-store-csi-driver-config.yaml:170]: 6 Gi is fine.

Comment (SD-13) [reply config/jobs/kubernetes-sigs/secrets-store-csi-driver/secrets-store-csi-driver-config.yaml]: Done!

**Pull request merged**